src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -134,12 +139,16 @@ public void setPeriod(String period) {
     public void visitJavadocToken(DetailNode ast) {
         String firstSentence = getFirstSentence(ast);
         final int endOfSentence = firstSentence.lastIndexOf(period);
-        if (endOfSentence == -1) {
-            if (!isOnlyInheritDoc(ast)) {
-                log(ast.getLineNumber(), MSG_SUMMARY_FIRST_SENTENCE);
-            }
+        final String summaryDoc = checkPeriodAtEnd(ast).trim();
```

Comment:
> .trim()

should be done inside method that return Summary

>`
String firstSentence = getFirstSentence(ast);
...
String summaryDoc = checkPeriodAtEnd(ast)
`

in terms of Check, first sentence and summary is the same.
It is not clear why you calculte it by two different methods.
`checkPeriodAtEnd` should be renamed to `getSummarySentence` as minimum.